docs/concepts/overview/working-with-objects/common-labels-annotations.md

```diff
@@ -0,0 +1,186 @@
+
+In addition to supporting tooling, the recommended labels and annotations describe
+applications, in a generic sense, in a way that can queried or are otherwise useful
+when operating applications.
+{% endcapture %}
+
+{% capture body %}
+The recommended object metadata is broken apart so that some of it is labels while
+other parts are annotations. Labels are shorter and can be used when querying
+the Kubernetes API. Annotations allow for longer data that's not meant to be
+queried.
+
+The metadata is organized around the concept of an Application. Kubernetes is not
+a Platform as a Service and doesn't have or enforce a formal notion of an application.
+Instead, applications are informal, described with metadata, and the definition of what an
+application contains is loose.
+
+Note, these are recommended labels and annotations. They aid the use of managing
+applications but are not required for any core tooling to work.
+
+One common theme you'll notice about the labels and annotations is that they
+contain a prefix of `app.kubernetes.io`. Labels and annotations without a prefix
+are considered private to users. These are meant to be common labels that do not
+interfere with existing or custom user labels. To accomplish this a prefix is
+used.
+
+## Labels
+
+| Key                                 | Description           | Example  | Type |
+| ----------------------------------- | --------------------- | -------- | ---- |
+| `app.kubernetes.io/name`              | The name of the application stack | `wordpress` | string |
+| `app.kubernetes.io/version`           | The current version of the running application (e.g., a semantic version, revision hash, etc.) | `v1.2.3` | string |
+| `app.kubernetes.io/component`         | The component within the architecture | `front-end` | string |
+| `app.kubernetes.io/component-name`    | The name of a component within the application | `mysql` | string |
+| `app.kubernetes.io/component-version` | The current version of the component | `5.7.21` | string |
+| `app.kubernetes.io/deploy-manager`    | The tool being used to manage the operation of an application | `helm` | string |
+
+To illustrative these labels in action consider the following StatefulSet object
+containing them:
+
+```yaml
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  labels:
+    app.kubernetes.io/name: wordpress
+    app.kubernetes.io/version: "4.9.4"
+    app.kubernetes.io/component: database
+    app.kubernetes.io/component-name: mysql
+    app.kubernetes.io/component-version: "5.7.21"
+    app.kubernetes.io/deploy-manager: helm
+```
+
+## Annotations
+
+| Key                     | Description           | Example  | Type |
+| ----------------------- | --------------------- | -------- | ---- |
+| `app.kubernetes.io/usage` | The location of an applications usage information (e.g., details for how it's being run by an organization). A preference is given to version specific information | `https://example.com/foo/1.2/` | URL |
```

Comment:
The application KEP has some other annotations such as LICENSE.  Have these been considered?

Would we want annotations for operational concerns for dashboards to display?

* Escalation point / pager (email / phone)
* Monitoring dashboard (URL)
* Logs dashboard (URL)
* Playbooks (URL)

Reply:
Those are happening on the [Application](https://github.com/kubernetes-sigs/application) sub-project of SIG Apps. The KEP has been moved to a sub-project. The labels and annotations have been looked at as complementary to the Application CRD.

A number of those items, such as escalation point, is something many store in a 3rd party service. Things that are specific like this are a bit debated.

Reply:
Where are the boundaries between fields in the Application CRD and annotations vs labels.  What defines one as belonging in one but not the other?  Is there overlap in the 2?